Tests/ClientRuntimeTests/NetworkingTests/ContentLengthMiddlewareTests.swift

```diff
@@ -0,0 +1,85 @@
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        builtContext = HttpContextBuilder()
+                  .withMethod(value: .get)
+                  .withPath(value: "/")
+                  .withEncoder(value: JSONEncoder())
+                  .withDecoder(value: JSONDecoder())
+                  .withOperation(value: "Test Operation")
+                  .build()
+        stack = OperationStack<MockInput, MockOutput, MockMiddlewareError>(id: "Test Operation")
+    }
+
+    func testTransferEncodingChunkedSetWhenStreamLengthIsNil() async throws {
+        addContentLengthMiddlewareWith(requiresLength: false, unsignedPayload: true)
+        forceEmptyStream()
+        try await AssertHeadersArePresent(expectedHeaders: ["Transfer-Encoding": "Chunked"])
+    }
+
+    func testContentLengthSetWhenStreamLengthAvailableAndRequiresLengthSet() async throws {
+        addContentLengthMiddlewareWith(requiresLength: true, unsignedPayload: false)
+        try await AssertHeadersArePresent(expectedHeaders: ["Content-Length": "0"])
+    }
+
+    func testContentLengthSetWhenRequiresLengthAndUnsignedPayload() async throws {
+        addContentLengthMiddlewareWith(requiresLength: true, unsignedPayload: true)
+        try await AssertHeadersArePresent(expectedHeaders: ["Content-Length": "0"])
+    }
+
+    func testRequiresLengthSetWithNilStreamShouldThrowError() async throws {
+        addContentLengthMiddlewareWith(requiresLength: true, unsignedPayload: false)
+        forceEmptyStream()
+        do {
+            try await AssertHeadersArePresent(expectedHeaders: ["Content-Length": "0"])
+            XCTFail("Should throw error")
+        } catch let error as StreamError {
+            switch error {
+            case .notSupported("sigv4 requires length"), .notSupported("operation requires length"):
+                // The error matches one of the expected cases, test passes
+                break
+            default:
+                XCTFail("Error is not StreamError.notSupported with expected message")
+            }
+        }
+    }
+
+    private func addContentLengthMiddlewareWith(requiresLength: Bool, unsignedPayload: Bool) {
+        stack.finalizeStep.intercept(
+            position: .before,
+            middleware: ContentLengthMiddleware(requiresLength: requiresLength, unsignedPayload: unsignedPayload)
+        )
+    }
+
+    private func forceEmptyStream() {
+        // Force stream length to be nil
+        stack.finalizeStep.intercept(position: .before, id: "set nil stream length") { (context, input, next) -> OperationOutput<MockOutput> in
+            input.body = .stream(BufferedStream()) // Set the stream length to nil
+            return try await next.handle(context: context, input: input)
+        }
+    }
+        
+    private func AssertHeadersArePresent(expectedHeaders: [String: String]) async throws -> Void {
```

Comment:
When creating a reusable test assertion, it is helpful to have it capture the file and line so that the test failure can be marked in the right place in the IDE window.

See this explanation: https://www.iosdev.recipes/xctest/custom-assertions-file-and-line/

Reply:
Updated function to pass file & line